src/traces/scattermapbox/attributes.js

```diff
@@ -26,6 +26,58 @@ module.exports = overrideAll({
     lon: scatterGeoAttrs.lon,
     lat: scatterGeoAttrs.lat,
 
+    cluster: {
+        enabled: {
+            valType: 'boolean',
+            role: 'info',
+            dflt: false,
+            editType: 'plot',
+            description: 'Determines whether clustering is enabled or disabled.'
+        },
+        maxzoom: {
+            valType: 'number',
+            role: 'info',
+            min: 0,
+            max: 22,
+            dflt: 14,
+            description: 'Sets the maximum zoom level for the cluster.',
+        },
+        radius: {
+            role: 'info',
+            valType: 'number',
+            dflt: 50,
+            description: 'Radius of each cluster when clustering points.',
+        },
+        step: {
+            role: 'info',
+            valType: 'number',
+            arrayOk: true,
+            dflt: -1,
+            min: -1,
+            description: [
+                'Sets the steps for each cluster.'
+            ].join(' ')
+        },
+        size: {
+            role: 'info',
+            valType: 'number',
+            arrayOk: true,
+            dflt: 20,
+            min: 0,
+            description: [
+                'Sets the size for each cluster step.'
+            ].join(' ')
+        },
```

Comment:
In addition to `cluster.color`, it would be nice to add `cluster.opacity` too. That way one could easily adjust transparency for those large clustered regions over the map.